packages/amplify-category-notifications/lib/pinpoint-helper.js

```diff
@@ -37,27 +36,34 @@ async function ensurePinpointApp(context) {
 }
 
 async function createPinpointApp(context) {
-  const { projectConfig, amplifyMeta } = context.exeInfo;
+  const { projectConfig, amplifyMeta, localEnvInfo } = context.exeInfo;
```

Comment:
Make actual calls to get the current amplifyMeta, projectConfig and localInfo. The context.exeInfo might not have the latest amplifyMeta when switching envs.